lib/src/views/post/edit_post_screen.dart

```diff
@@ -659,64 +668,79 @@ class _EditPostScreenState extends State<EditPostScreen> {
-                                              null &&
-                                          attachments!.isNotEmpty) &&
-                                      mapIntToMediaType(attachments!
-                                              .first.attachmentType) ==
-                                          MediaType.link &&
-                                      showLinkPreview) ||
-                                  (linkModel != null && showLinkPreview)
-                              ? Stack(
-                                  children: [
-                                    LMLinkPreview(
-                                      linkModel: linkModel,
-                                      backgroundColor:
-                                          theme!.colorScheme.surface,
-                                      showLinkUrl: false,
-                                      onTap: () {
-                                        launchUrl(
-                                          Uri.parse(
-                                              linkModel?.ogTags?.url ?? ''),
-                                          mode: LaunchMode.externalApplication,
-                                        );
-                                      },
-                                      border: const Border(),
-                                      title: LMTextView(
-                                        text: linkModel?.ogTags?.title ?? "--",
-                                        maxLines: 1,
-                                        overflow: TextOverflow.ellipsis,
-                                        textStyle: theme!.textTheme.titleMedium,
-                                      ),
-                                      subtitle: LMTextView(
-                                        text: linkModel?.ogTags?.description ??
-                                            "--",
-                                        maxLines: 2,
-                                        overflow: TextOverflow.ellipsis,
-                                        textStyle:
-                                            theme!.textTheme.displayMedium,
-                                      ),
-                                    ),
-                                    Positioned(
-                                      top: 5,
-                                      right: 5,
-                                      child: GestureDetector(
+                    if (postDetails!.isRepost && showRepost)
+                      SliverToBoxAdapter(
+                        child: NovaRepostWidget(
+                          post: PostViewModel.fromPost(
+                              post: repostedPosts![postDetails!.attachments!
+                                  .first.attachmentMeta.entityId]!),
+                          user: user!,
+                        ),
+                      ),
+                    if (!showRepost)
```

Comment:
can we access repost widget using attachments list instead of defining a diff widget

Reply:
it is a check to not build and add link type attachments in edit post